### ISSUE TITLE
New GUI only works with mouse, whilst OSD menu worked also with controller

### ISSUE THREAD

**Issue by MBilderbeek** [openMSX/openMSX#1666]
Strictly speaking, this is a regression: the old OSD menu allows you to control openMSX with a controller (as well as the mouse and keyboard). But the new Dear ImGUI based GUI only supports mouse so far.

This issue is to document that it's worth it to look at supporting controller/keyboard for the new GUI as well.

**Comment by MBilderbeek** [comment]
There's a way to enable controller operatoin (e.g see.  ImGuiManager.cc:144), but it interferes with emulated MSX controllers. 
The suggestion is to use a dedicated controller button to toggle between "MSX controller" and "openMSX controller" mode. Seems OK to me.

**Comment by MBilderbeek** [comment]
I tried to enable that option (currently on line 133 of ImGuiManager.cc), and this is what I found in a first test:
- button 1 opens the same menu that you get with CTRL-TAB. It shows 2 lines: (Untitled) and (Main Menu Bar). (The former is probably the status bar?)  However, I see no way to select between them and thus no way to activate the Main Menu Bar. When I close the Status Bar (with the setting), I *can* activate the Main Menu Bar and actually control it with the analog stick and button 2 (select) and button 3 (cancel). With some playing around, I am able to even close windows: by pressing button 1 in an active window, it allows me to select the top left and top right (x) icons of the window handle, so I can close the window.
- when playing a game with joystick, I use button 1 of course to fire (e.g. in Space Manbow), but when I keep it pressed too long, the CTRL-TAB window selection menu is visible as long as I press it.

So, basically, two issues need to be solved:
- button 1 interferes with our bound button 1. (Maybe we can change the button binding for its functionality?)
- there doesn't seem to be a way to select the window in the window selector menu ("CTRL-TAB menu") - but this may be related to the other point.

Other than that, it seems to be pretty usable for the most part, although it does feel a bit clumsy.

**Comment by MBilderbeek** [comment]
@ocornut Anything you can recommend for our situation? Are we doing something wrong? How to avoid interference of "MSX" controllers?

**Comment by ocornut** [comment]
There’s a full description of gamepad controls in the doc and a PSD sheet. 

> However, I see no way to select between them and thus no way to activate the Main Menu Bar.

L1/R1 buttons.

Windows may be excluded from the list.
Depending on your UI you may want to hardcode a shortcut to eg access the main menu.

> when playing a game with joystick, I use button 1 of course to fire (e.g. in Space Manbow), but when I keep it pressed too long, the CTRL-TAB window selection menu is visible as long as I press it.

Well the first question to ask is how would you like this to work, from the users point of view, ideally?

While keyboard Ctrl-tab is remappable, the gamepad one isn’t yet because multiple inputs are layered (tap on left face button toggle between main and menu layer, hold left face button to activate windowing to focus a window but also move/resize them). Depending on what would be your ideal mapping we can figure out a solution.

The simplest way would be indeed to have a single gamepad button to switch between game inputs and openmsx/ImGui inputs and then it is trivial for you.
